### ISSUE TITLE
Crash on exit when Xwayland is not found

### ISSUE THREAD

**Issue by maringuu** [ifreund/river#167]
First of all it might be related to my system (musl + gcompat, river compiled in an arch chroot). I'll try the `zig-wlroots`branch in the future. 

Steps to reproduce:
1. Have river compiled with `-Dxwayland`
2. Remove Xwayland from your system
3. Start river with an emtpy config
4. exit river

Somehow river crashes with the following log.

Log: https://paste.sr.ht/~maringuu/d7ca0e7912573cdacd239cc884be066550878a73
The relevant part is
```
00:00:01.554 [ERROR] [xwayland/server.c:234] Xwayland startup failed, not setting up xwm
debug: (keyboard) switch VT keysym received
notice: (server) switching to VT 1
00:00:04.280 [INFO] [backend/drm/backend.c:103] DRM fd paused
info: (server) shutting down
Segmentation fault at address 0x8
???:?:?: 0x7fb77b22f77b in ??? (???)
integer overflow
/home/maringuu/opt/zig-linux-x86_64-0.7.0/lib/zig/std/debug.zig:141:69: 0x28c7a5 in std.debug.dumpStackTraceFromBase (river)
            printSourceAtAddress(debug_info, stderr, return_address - 1, tty_config) catch return;
                                                                    ^
/home/maringuu/opt/zig-linux-x86_64-0.7.0/lib/zig/std/debug.zig:1782:35: 0x279265 in std.debug.handleSegfaultLinux (river)
            dumpStackTraceFromBase(bp, ip);
```

**Comment by travankor** [comment]
It still aborts on the latest commit + glibc.

**Comment by ifreund** [comment]
Fixed by 9e3e92050e04320949c6cd995273c30319ebd515
